api/pkg/db/initializer/initializer.go

```diff
@@ -80,6 +81,40 @@ func (i *Initializer) Run(ctx context.Context) (*model.Config, error) {
 
 type initFn func(*gorm.DB, *log.Logger, *app.Data) error
 
+// Deletes categories and it's mapping with resource from the database
+// which are removed from the config
+func deleteCategories(db *gorm.DB, log *log.Logger, data *app.Data) error {
```

Comment:
Instead of doing this separately can we
- after adding the category https://github.com/tektoncd/hub/blob/49e68f837eb3a9f36b9fdbc675bb4d5167c090c1/api/pkg/db/initializer/initializer.go#L87
after this take the id of the category and put in an array
-  so this will add id of all categories which are in the config yaml
- after this delete categories whose id is not in the above array so it will done in a single query

Reply:
+1

Reply:
but we need  to delete resource & category mapping before deleting categories

Reply:
yeah can we not use the same resource ids to delete the mapping
delete mapping if resource id not in the array?

Reply:
yeah we can, but number of query will be increase

Reply:
```
// Not In
db.Not(map[string]interface{}{"name": []string{"jinzhu", "jinzhu 2"}}).Delete(&User{})
// Delete * FROM users WHERE name NOT IN ("jinzhu", "jinzhu 2");
```
wouldn't something like this work?